src/core/public/chrome/nav_links/nav_links_service.ts

```diff
@@ -154,25 +161,28 @@ export class NavLinksService {
           return;
         }
 
-        navLinks$.next(new Map([...navLinks$.value.entries()].filter(([linkId]) => linkId === id)));
+        const updater: LinksUpdater = navLinks =>
+          new Map([...navLinks.entries()].filter(([linkId]) => linkId === id));
+
+        linkUpdaters$.next([...linkUpdaters$.value, updater]);
       },
 
       update(id: string, values: ChromeNavLinkUpdateableFields) {
         if (!this.has(id)) {
           return;
         }
 
-        navLinks$.next(
+        const updater: LinksUpdater = navLinks =>
           new Map(
-            [...navLinks$.value.entries()].map(([linkId, link]) => {
+            [...navLinks.entries()].map(([linkId, link]) => {
               return [linkId, link.id === id ? link.update(values) : link] as [
                 string,
                 NavLinkWrapper
               ];
             })
-          )
-        );
+          );
 
+        linkUpdaters$.next([...linkUpdaters$.value, updater]);
```

Comment:
Before PR, we were reading the `application.availableApps` only once at startup, which allowed to directly edit the links in `NavlinkServices`. As `availableApps$` is now an Observable, I have to keep track of every 'manually' applied changes in `NavlinkServices` to reapply them after each `availableApps$` emission. Hopefully at some point, we will no longer allow direct edition to Navlinks, and this will disappear, but for now, this seems like a necessary evil for retro-compat.